### DISCUSSION TITLE
Multiple ISO cycling - specify block size ?

### DISCUSSION THREAD

**Discussion by vom513** [ZuluSCSI/ZuluSCSI-firmware#266] [Q&A]
Hello,

From the example ini:

```
# Settings can be overridden for individual devices.
#[SCSI2]
#Product = "Disk with ID2"

#[SCSI5]
#Product = "CD-ROM Drive"
#Type = 2

# If IMG0..IMG9 are specified, they are cycled after each CD eject command.
#IMG0 = FirstCD.iso
#IMG1 = SecondCD.bin
```

Makes sense.  However - is there a way to also specify block size in the filename while using IMG0/1/.. ?  I use my ZSes mainly on Sun machines that need 512 block size.

Thanks.

**Comment by vom513** [comment]
My curiosity got the best of me so I tried it out.  It worked, but I don't think the block size was set properly (assuming the log is accurate):

```
275ms] -- Opening 'solaris-9-0502-sparc-install_512.iso' for id:6, specified in zuluscsi.ini
[276ms] ---- Read prefetch enabled: 8192 bytes
[278ms] Finding HDD images in directory /:
[279ms] -- Opening /HD30_512-9gb-sol-BLANK.hda for id:3 lun:0
[280ms] ---- Read prefetch enabled: 8192 bytes
[282ms] SCSI ID:3 BlockSize:512 Type:0 Quirks:0 ImageSize:9437184kB
[283ms] SCSI ID:6 BlockSize:2048 Type:2 Quirks:0 ImageSize:622060kB
```

As you can see when IMG0 is attached, it's coming in as 2048 block.  Here is my ini:

```
[SCSI6]
Product = "ZS CD-ROM"
Type = 2

# If IMG0..IMG9 are specified, they are cycled after each CD eject command.
IMG0 = solaris-9-0502-sparc-install_512.iso
IMG1 = solaris-9-0502-sparc-software-1of2_512.iso
IMG2 = solaris-9-0502-sparc-software-2of2_512.iso
IMG3 = solaris-9-0502-sparc-supplemental_512.iso
```

I'm surprised that the system booted and installed with no errors.

Full log attached as well.

[iso_cycle-zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/12134954/iso_cycle-zululog.txt)

**Comment by vom513** [comment]
Sorry to keep replying to myself - but some further experimentation and I've come up with this (posted to an old Suns mailing list, pasting out of laziness...):

- CD6.iso : Zulu log shows it coming up in 2048-byte mode, but my sun4c/m machines boot fine from it.  I don’t see a MODE SELECT in the log - I guess the ZS is just operating in “soft” block mode and honoring whatever request comes in ? (This is great by the way - not looking a gift horse in the mouth)

- CD6_512.iso : Block size specified as 512.  Works as expected.

- CD6_2048.iso : Block size specified as 2048.  Machines will NOT boot.  

So it seems like _<BLOCKSIZE> is a “hard” setting.  Omitting this seems to run the device in “soft” mode (hard/soft just my personal terminology for this behavior).

**Reply by lausvi** [reply]
Just to chime in; I was wondering this very same thing and found myself here. I needed to have a 512 block size CD-ROM with cycling (to install AIX 5.1) and it seems that the file names can be anything (or at least it doesn't make any difference under the cycling) and the image just worked as 512 block size.

The instructions could use some further details on how to attach the first image (not as a drive with CD4_512... like usually) but just listed under the IMG0, IMG1... list.

Also looking at the log (trying to figure this out) it appears that the first press of the Eject button unmounts the previous image, and only the second press attaches the next one?
